acq4/util/imaging/sequencer.py

```diff
@@ -240,7 +240,7 @@ def handle_new_frames(f: "Frame | list[Frame]", idx: int):
                 for move in movements_to_cover_region(imager, mosaic):
                     _future.waitFor(move)
                     if z_stack:
-                        stack = acquire_z_stack(imager, *z_stack, block=True).getResult()
```

Comment:
@campagnola here's the problematic call. we can't make it a separate thread, or the device locking won't work. we can't rely on `stop()` on the parent, since it's not in a `checkStop` loop at this point.